trikot-streams/coroutines-interop/src/commonTest/kotlin/kotlinx/coroutines/reactive/FlowAsPublisherTest.kt

```diff
@@ -78,6 +91,23 @@ class FlowAsPublisherTest : TestBase() {
         finish(3)
     }
 
+
+    @Test
+    fun testFlowWithSharedResultIsShared() = runTest {
+        val publisher = flowOf(1)
+            .map { n ->
+                expect(1)
```

Comment:
Run the expect in the subscription (onEvent), not in the map

Reply:
Actually, I'm testing that we do share the value after a second subscribe instead of reprocessing the value in the map.

Reply:
Ah gotcha, it does make sense!